File: pptx-service/renderers/checklist.py
from builder import add_bg, add_textbox, add_rect, add_eyebrow, add_line
from theme import ThemeColors


def render(slide, data: dict, t: ThemeColors) -> None:
    add_bg(slide, t.bg)
    heading = data.get("heading", "")
    items = data.get("items", [])
    eyebrow = data.get("eyebrow", "")

    if eyebrow:
        add_eyebrow(slide, eyebrow, t)
    add_textbox(slide, 0.83, 0.7, 11.0, 1.2, heading,
                t.font_display, 48, t.text, bold=True)
    add_line(slide, 0.83, 2.0, 12.5, 2.0, t.border, 0.5)

    y = 2.2
    for item in items[:7]:
        add_rect(slide, 0.83, y + 0.05, 0.42, 0.42,
                 line_hex=t.accent, line_width_pt=1.5)
        add_textbox(slide, 1.4, y, 11.0, 0.6, item,
                    t.font_body, 28, t.text)
        y += 0.75

File: lib/pptx/slides/checklist.ts
import type { ChecklistSlide } from '@/lib/types';
import {
  px, py, SLIDE_W,
  setBackground, addRect, addTextBox, addHeading, addPageNumber,
  type PptxTheme,
} from '../helpers';
import type PptxGenJS from 'pptxgenjs';
type PSlide = ReturnType<InstanceType<typeof PptxGenJS>['addSlide']>;

export function render(slide: PSlide, data: ChecklistSlide, t: PptxTheme, n: number, total: number): void {
  setBackground(slide, '#' + t.bg);
  addHeading(slide, data.heading, t, py(80));

  data.items.forEach((item, i) => {
    const y = py(220) + i * py(110);
    addRect(slide, t.paddingIn, y, px(28), py(38), '#' + t.bg, { line: { color: t.accent, width: 2 } });
    addTextBox(slide, item, {
      x: t.paddingIn + px(48), y, w: SLIDE_W - 2 * t.paddingIn - px(48),
      fontSize: t.bodySize, color: '#' + t.text, fontFace: t.fontBody, wrap: true,
    });
  });

  addPageNumber(slide, n, total, t);
}
